authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/validator/SAMLFileUploadValidator.java

```diff
@@ -0,0 +1,92 @@
+package com.synopsys.integration.alert.authentication.saml.validator;
+
+import com.synopsys.integration.alert.api.common.model.ValidationResponseModel;
+import com.synopsys.integration.alert.common.descriptor.config.field.validation.ValidationResult;
+import com.synopsys.integration.alert.common.persistence.util.FilePersistenceUtil;
+import com.synopsys.integration.alert.component.authentication.descriptor.AuthenticationDescriptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Component;
+import org.xml.sax.ErrorHandler;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+import org.xml.sax.SAXParseException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.function.Function;
+
+@Component
+public class SAMLFileUploadValidator {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final FilePersistenceUtil filePersistenceUtil;
+
+    @Autowired
+    public SAMLFileUploadValidator(FilePersistenceUtil filePersistenceUtil) {
+        this.filePersistenceUtil = filePersistenceUtil;
+    }
+
+    public ValidationResponseModel validateMetadataFile(Resource resource) {
+        return validateFile(
+            AuthenticationDescriptor.SAML_METADATA_FILE,
+            resource,
+            this::validateXMLFile
+        );
+    }
+
+    public ValidationResponseModel validateFile(String fileName, Resource resource, Function<File, ValidationResult> validateFunction) {
```

Comment:
Can this function be added to the FileUploadHelper.  The certificate files will need to do the same thing for validating.   Create a temp file and then test validation for the specific file.

Reply:
Resolved, made the method private instead as it will be reused in this file for certs validation